test/integration/src/test/java/io/pravega/test/integration/StreamMetricsTest.java

```diff
@@ -229,4 +229,50 @@ public void testTransactionMetrics() throws Exception {
         AssertExtensions.assertEventuallyEquals(true, () -> MetricRegistryUtils.getCounter(MetricsNames.ABORT_TRANSACTION, streamTags(txScopeName, txStreamName)) != null, 10000);
         assertEquals(1, (long) MetricRegistryUtils.getCounter(MetricsNames.ABORT_TRANSACTION, streamTags(txScopeName, txStreamName)).count());
     }
+
+    @Test(timeout = 30000)
+    public void testRollingTxnMetrics() throws Exception {
+        String scaleRollingTxnScopeName = "scaleRollingTxnScope";
+        String scaleRollingTxnStreamName = "scaleRollingTxnStream";
+
+        controllerWrapper.getControllerService().createScope(scaleRollingTxnScopeName).get();
+        if (!controller.createStream(scaleRollingTxnScopeName, scaleRollingTxnStreamName, config).get()) {
+            log.error("Stream {} for scale testing already existed, exiting", scaleRollingTxnScopeName + "/" + scaleRollingTxnStreamName);
+            return;
+        }
+
+        @Cleanup
+        EventStreamClientFactory clientFactory = EventStreamClientFactory.withScope(scaleRollingTxnScopeName, ClientConfig.builder()
+                .controllerURI(URI.create("tcp://localhost:" + controllerPort)).build());
+        @Cleanup
+        TransactionalEventStreamWriter<String> writer = clientFactory.createTransactionalEventWriter(Stream.of(scaleRollingTxnScopeName, scaleRollingTxnStreamName).getStreamName(),
+                new JavaSerializer<>(), EventWriterConfig.builder().build());
+        Transaction<String> transaction = writer.beginTxn();
+        transaction.writeEvent("Transactional content");
+
+        //split to 3 segments
+        Map<Double, Double> keyRanges = new HashMap<>();
+        keyRanges.put(0.0, 0.25);
+        keyRanges.put(0.25, 0.75);
+        keyRanges.put(0.75, 1.0);
+
+        Stream scaleRollingTxnStream = new StreamImpl(scaleRollingTxnScopeName, scaleRollingTxnStreamName);
+        if (!controller.scaleStream(scaleRollingTxnStream, Collections.singletonList(0L), keyRanges, executor).getFuture().get()) {
+            log.error("Scale stream: splitting segment into three failed, exiting");
+            return;
+        }
+
+        assertEquals(3, (long) MetricRegistryUtils.getGauge(MetricsNames.SEGMENTS_COUNT, streamTags(scaleRollingTxnScopeName, scaleRollingTxnStreamName)).value());
+        assertEquals(1, (long) MetricRegistryUtils.getGauge(MetricsNames.SEGMENTS_SPLITS, streamTags(scaleRollingTxnScopeName, scaleRollingTxnStreamName)).value());
+        assertEquals(0, (long) MetricRegistryUtils.getGauge(MetricsNames.SEGMENTS_MERGES, streamTags(scaleRollingTxnScopeName, scaleRollingTxnStreamName)).value());
+
+        transaction.flush();
+        transaction.commit();
+
+        Thread.sleep(1000);
```

Comment:
This sleep may turn the test into a new flaky one.
You could wait for the conditions below to be verified

Reply:
+1. Please do not add arbitrary thread sleeps in tests

Reply:
Use `TestUtils.await` or `AssertExtensions.assertEventuallyEquals`

Reply:
fixed, sorry for the overlooking, the extensions are being used already in the rest of the class.